### ISSUE TITLE
mRemoteNG to front

### ISSUE THREAD

**Issue by gschmitt99** [mRemoteNG/mRemoteNG#398]
When the mRemoteNG window is behind another but still partially visible, and click on the client area of mRemoteNG, it is not brought to the front.  There are times it does, and times it does not.  As I write this description this morning, I am messing with multiple tabs open in Chrome, clicking between Chrome and mRemoteNG and different tabs in Chrome.  Sometimes the behavior does not manifest, and sometimes it does.

|--:|---|
|Operating system | Windows 7 x64 |
|mRemoteNG version| 1.75 aplha 3  |

**Comment by kmscode** [comment]
Most likely when you encounter this problem you have an SSH session open I assume?

There are known focus issues with PuTTY sessions due to the way it's implemented...

Reference #290

**Comment by gschmitt99** [comment]
Yes, I am using SSH.  I'm not familiar with how this project is setup, so is putty actually being used somewhat like a plugin into mRemoteNG?

My issues are even simpler than was was described in 290.  I don't have to switch tabs or anything, just between windows.

**Comment by kmscode** [comment]
Putty is a separate process that we just stick inside of the mRemoteNG window. This causes focus issues.

If you click in the putty window, that doesn't appear to be registered to the parent. 

Most likely the solution will be #181

**Comment by bluikko** [comment]
MTPuTTY does not have this issue (but it has its own set of other problems) - from the user point of view both mRemoteNG and MTPuTTY seem to work the same way.

To me this (Alt-Tab / clicking to switch tasks in/out of mRemoteNG is not working) and #196 are huge annoyances that disrupt efficient working.

If I would know how much fixing these features could cost I could check with my company if it would be possible to sponsor. Several thousand USD would probably be too much, tho.

I could even entertain opening some crowdfunding project for fixing these 2 issues - something I would not normally even think about. mRemoteNG just seems like otherwise perfect solution for us.

**Comment by Kvarkas** [comment]
Video proof: https://youtu.be/DRRC8HvW0Ts

**Comment by robertpopa22** [comment]
Triage refresh (P4 version-label debt): this issue is tagged with legacy version labels. Please retest on the latest build (fork v1.79.x or upstream dev), then share reproducible steps and expected vs actual behavior. If still valid, keep open and relabel to current triage scope (for example Need 2 check). If fixed, close with resolution.

Current legacy version labels: 1.77.3

**Comment by robertpopa22** [comment]
**Fix available for testing**

A fix for this issue has been implemented in our community fork (`robertpopa22/mRemoteNG`, branch `main`). A beta build including this fix is available from the fork's Releases page.

Please test and report if this resolves your issue.

---

> *We sincerely appreciate the mRemoteNG maintainers and the wonderful community that has kept this project alive and thriving for over a decade. Your work powers thousands of IT professionals worldwide, including us. Our fork builds entirely on your foundation — we're preparing a consolidated Pull Request to contribute these improvements back to the official project. Thank you for everything you've built.*
>
> *— robertpopa22, community contributor*
